File: pos_bahrain/doc_events/item_price.py
# -*- coding: utf-8 -*-
# Copyright (c) 2019, 	9t9it and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe

from pos_bahrain.api.item import get_conversion_factor


def before_save(doc, method):
    doc.pb_conversion_factor = get_conversion_factor(doc.item_code, doc.uom)
    if doc.customer and not doc.pb_customer_name:
        doc.pb_customer_name = frappe.db.get_value(
            "Customer", doc.customer, "customer_name"
        )

File: pos_bahrain/public/js/scripts/item_price.js
export default {
  setup: function(frm) {
    frm.set_query('uom', function({ item_code }) {
      return {
        query: 'pos_bahrain.api.item.query_uom',
        filters: { item_code },
      };
    });
  },
  uom: async function(frm) {
    const { item_code, uom } = frm.doc;
    if (uom) {
      const { message: pb_conversion_factor } = await frappe.call({
        method: 'pos_bahrain.api.item.get_conversion_factor',
        args: { item_code, uom },
      });
      frm.set_value({ pb_conversion_factor });
    } else {
      frm.set_value('pb_conversion_factor', null);
    }
  },
  customer: function(frm) {
    if (!frm.doc.customer) {
      frm.set_value('pb_customer_name', null);
    }
  },
};
